### ISSUE TITLE
[BUG] Not possible to log in to github

### ISSUE THREAD

**Issue by lrustand** [zeapo/Android-Password-Store#2393]
### Describe the bug

It is not possible to log in to github.com using account password in order to clone the password store. I have tried at least 20 times.

### Steps to reproduce

1. Click on Clone remote repo
2. Type in a github.com repo
3. Select password authentication
4. Type in correct password
5. App says wrong password

### Expected behavior

Expect to be able to log in.

### Device information

- Device: Samsung Galaxy S22
- OS: Android 13
- App version: 1.13.5

**Comment by msfjarvis** [comment]
If you use two factor authentication you cannot use your regular password to authenticate with GitHub, create a personal access token and use that as a password.

**Comment by lrustand** [comment]
Yes I have tried that also. I found a similar issue https://github.com/android-password-store/Android-Password-Store/issues/1071, and tried following the instructions there, but I keep getting 'wrong password' even when using personal access token.

**Comment by msfjarvis** [comment]
Did you use the correct URL format given in the issue? I use HTTPS w/ a GitHub private repo daily so I know for a fact that it definitely works 🤷🏼

**Comment by lrustand** [comment]
Yes, I am using *REDACTED*

**Comment by lrustand** [comment]
I have also tested the personal access token by using git clone, and can confirm that it can clone the repository on my computer, so the access token has the correct permission. The repository exists and is not empty, but is private.

**Comment by lrustand** [comment]
Why is this "waiting on author"? I am waiting for you. I have followed every step correctly, but it does not work. If you need more information or there are more hidden requirements please tell me, I have given you all information I have and done everything I can. Also "needs reproduction steps" is wrong, I have given them in the OP.

**Comment by msfjarvis** [comment]
If you attempt to follow the timeline of the issue, you will observe that the "waiting on author" label was applied prior to your reply. Ergo, it was accurate at the time it was used and clearly means I have not re-triaged the issue since.

The "needs reproduction steps" label is present because I can not reproduce the issue with your provided steps, since all you've done is outlined a very basic workflow that I use on a daily basis and have already said works for me.

Now that we've addressed your unwarranted impatience over support you are not owed and I am not obliged to provide, let's try something more productive. Follow the steps [here](https://docs.passwordstore.app/docs/users/reporting-bugs/) to capture a log that may give me more hints as to what's happening on your device, and either email it to the address on that page or attach it with a comment to this issue.

I'll be waiting.

**Comment by lrustand** [comment]
Thank you. I have sent the log to your email.

**Comment by msfjarvis** [comment]
> Thank you. I have sent the log to your email.

There was a DNS configuration error on the passwordstore.app domain that caused email delivery to be broken, I've fixed it but in the mean time please forward the email to me@msfjarvis.dev

**Comment by msfjarvis** [comment]
@lrustand the email you sent me does not have seem to have the attachment preserved so there are no logs for me to see.

**Comment by msfjarvis** [comment]
Got your follow up email but it doesn't look like you enabled debug logging in Password Store, the entire log you sent is just Samsung spam.

**Comment by lrustand** [comment]
Yes, I noticed that the log looked pretty useless. I did however follow the full instructions you linked, including enabling debug logging, you can see in the attached screenshot that debug logging is actually enabled. Maybe it did not take effect or the app did not fully restart, let me try again. I have sent you a new email.
![Screenshot_20230303_173105_Password Store](https://user-images.githubusercontent.com/13719791/222775388-a5fc2c5d-3747-4239-aca6-04cf11d68d3a.jpg)

**Comment by msfjarvis** [comment]
Nothing particularly helpful in the new one either, can you try the [snapshot build](https://docs.passwordstore.app/docs/users/release-channels/)? It can be installed alongside the 1.x.y release.

**Comment by lrustand** [comment]
Sent you more logs using snapshot build.

**Comment by msfjarvis** [comment]
```
03-03 19:02:34.866 20540 20540 D PasswordStore: org.eclipse.jgit.errors.TransportException: <REDACTED_GIT_URL>: not authorized
03-03 19:02:34.866 20540 20540 D PasswordStore: 	at org.eclipse.jgit.transport.TransportHttp.connect(SourceFile:280)
03-03 19:02:34.866 20540 20540 D PasswordStore: 	at org.eclipse.jgit.transport.TransportHttp.openFetch(SourceFile:3)
03-03 19:02:34.866 20540 20540 D PasswordStore: 	at androidx.compose.ui.node.NodeChain.executeImp(SourceFile:8)
03-03 19:02:34.866 20540 20540 D PasswordStore: 	at org.eclipse.jgit.transport.Transport.fetch(SourceFile:128)
03-03 19:02:34.866 20540 20540 D PasswordStore: 	at org.eclipse.jgit.api.FetchCommand.call(SourceFile:19)
03-03 19:02:34.866 20540 20540 D PasswordStore: 	at org.eclipse.jgit.api.PullCommand.call(SourceFile:34)
03-03 19:02:34.866 20540 20540 D PasswordStore: 	at app.passwordstore.util.git.GitCommandExecutor$execute$2$result$1.invokeSuspend(Unknown Source:7)
03-03 19:02:34.866 20540 20540 D PasswordStore: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(SourceFile:9)
03-03 19:02:34.866 20540 20540 D PasswordStore: 	at kotlinx.coroutines.DispatchedTask.run(SourceFile:109)
03-03 19:02:34.866 20540 20540 D PasswordStore: 	at kotlinx.coroutines.internal.LimitedDispatcher.run(SourceFile:12)
03-03 19:02:34.866 20540 20540 D PasswordStore: 	at kotlinx.coroutines.scheduling.TaskImpl.run(SourceFile:3)
03-03 19:02:34.866 20540 20540 D PasswordStore: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(SourceFile:77)
```

The Git library says your password is wrong, you should double check what you're using 🤷🏼

I just set up a fresh instance of the app with my personal pass repository, using a token generated by following the [docs for fine-grained tokens](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-fine-grained-personal-access-token) and things work perfectly fine.

**Comment by lrustand** [comment]
Yes, the app also says the password is wrong, but it works on my computer. I will admit that I only hand typed the access token the first three times, every time after this I have copied and pasted from the last hand typed one, however I did check it several times to see that it was correct. I will try a couple more times with manual typing again and be very careful to triplecheck every single character.

**Comment by lrustand** [comment]
No, it still claims that the "password" (access token) is wrong, even though I am 110% certain I entered it correctly this time.

**Comment by lrustand** [comment]
And I tested it once more with curl on my computer and it still works fine there.

**Comment by lrustand** [comment]
Btw, the log you are quoting from was me trying three times to log in, but it only showed up in the log the last time. The first two times it said "wrong password" like usual, but the third time it said something like "Not authorized".

**Comment by msfjarvis** [comment]
The only relevant possibilities left are as follows

1. You've typo'd the access token: The advice I can give for that is to not hand-type the token and instead paste it from your mobile browser into the app or use [scrcpy](https://github.com/Genymobile/scrcpy) to paste it from your computer.

2. Your access token is lacking the right permissions: For a fine-grained token, make sure the correct repository is selected, and you've granted read and write access to repository contents. For a legacy/classic token, make sure the entire `repo` scope is granted.

All evidence so far suggests this is a user error and not a bug with the app so I will move this to a repository discussion and raise a separate issue if needed.
